cdap-api/src/main/java/co/cask/cdap/api/lineage/Read.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Represents READ operation from data source into the collection of fields.
+ */
+public class Read extends Operation {
+  private final EndPoint endPoint;
+  private final Set<String> fields;
+
+  private Read(String name, String description, EndPoint endPoint, Set<String> fields) {
+    super(name, description);
+    this.endPoint = endPoint;
+    this.fields = Collections.unmodifiableSet(new HashSet<>(fields));
+  }
+
+  /**
+   * Get the EndPoint representing the data source for this READ operation
+   * @return the EndPoint
+   */
+  public EndPoint getEndPoint() {
+    return endPoint;
+  }
+
+  /**
+   * Get the set of fields created by this READ operation
+   * @return the set of fields
+   */
+  public Set<String> getFields() {
+    return fields;
+  }
+
+  /**
+   * Builder for Read operation
+   */
+  public static class Builder extends Operation.Builder {
+    private EndPoint endPoint;
+    private final Set<String> fields;
+
+    public Builder(String name) {
+      super(name);
+      this.fields = new HashSet<>();
+    }
+
+    /**
+     * Set the data source associated with the READ operation.
+     * @param endPoint the {@link EndPoint} representing the data source
+     * @return this
+     */
+    public Builder from(EndPoint endPoint) {
+      this.endPoint = endPoint;
+      return this;
+    }
+
+    /**
+     * Set the fields generated as a part of READ operation.
+     * @param fields the fields generated by READ operation
+     * @return this
+     */
+    public Builder to(String ... fields) {
```

Comment:
how do you ensure that the fields do not contain duplicates?

Reply:
I guess we can convert it to LinkedHashSet and then back to Array?